### DISCUSSION TITLE
SplitByName usage in python script

### DISCUSSION THREAD

**Discussion by krb0073** [GrossfieldLab/loos#67] [Q&A] [answered]
Hey Everyone, 
I had a quick question about the splitByName function from AtomicGroup. When playing around with the other splitBy* functions they all return vectors grouped by the desired subset. When I use splitByName, the only thing returned is a Swig Object. This happens no matter what selections give loos.selectAtoms() in python. I was wondering what the proper way is to use splitByName function?

Thanks for your help,
Kyle

**Comment by agrossfield** [comment]
Hi Kyle,

I'd actually forgotten `AtomicGroup.splitByName()` even exists -- oops! I just checked the docs (http://grossfieldlab.github.io/loos/classloos_1_1_atomic_group.html), and the return type of the other splitBy methods in c++ is a vector<AtomicGroup>, which gets turned into a list of AtomicGroup's in our SWIG wrapper. The return type of `splitByName` is `std::map< std::string, AtomicGroup >` -- my guess is we forgot to include a manual entry in the swig file to copy from a c++ map to a python dictionary, which I guess would be the most logical thing.

I just opened Issue #68 to deal with it. I can't look at it right now, but I'll try to make sure either @tromo or I take a look this week.

Alan

**Reply by agrossfield** [reply]
I created a pull request (#69) that I think fixes the problem. We should merge it in the next day or so. Once that's done, you'll need to pull down the latest version of the main branch with `git pull`, then rebuild loos with `scons PREFIX=$CONDA_PREFIX -j8 install`. At that point, it should work.

One thing I should have asked before: what are you trying to do with it? Part of why it never got exported to swig before is that we usually just use selections to accomplish the same things

**Comment by agrossfield** [comment, marked as answer]
Pull request merged.
